### PR TITLE
feat!: v19 BIP9 fork

### PR DESCRIPTION
**Pull request by ogabrielides**
## Issue being fixed or feature implemented

Added v19 BIP9 style fork structure along with utility functions. Since several features coming depending on that fork status, we needed to group them into one

## Checklist:

- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added or updated relevant unit/integration/functional/e2e tests
- [ ] I have made corresponding changes to the documentation

**For repository code-owners and collaborators only**
- [ ] I have assigned this pull request to a milestone

### CODE REVIEW

File: src/chainparams.cpp [hunk 1]
```diff
@@ -249,6 +249,15 @@ class CMainParams : public CChainParams {
         consensus.vDeployments[Consensus::DEPLOYMENT_DIP0024].nThresholdMin = 2420;   // 60% of 4032
         consensus.vDeployments[Consensus::DEPLOYMENT_DIP0024].nFalloffCoeff = 5;      // this corresponds to 10 periods
 
+        // Deployment of Deployment of Basic BLS, AssetLock, EHF
+        consensus.vDeployments[Consensus::DEPLOYMENT_V19].bit = 8;
+        consensus.vDeployments[Consensus::DEPLOYMENT_V19].nStartTime = 19999999999;     // TODO: To be determined later
+        consensus.vDeployments[Consensus::DEPLOYMENT_V19].nTimeout = 999999999999ULL;
+        consensus.vDeployments[Consensus::DEPLOYMENT_V19].nWindowSize = 100;
```
Comment (knst) [inline src/chainparams.cpp:256]: `nWindowsSize` should be 4032

File: src/chainparams.cpp [hunk 2]
```diff
@@ -249,6 +249,15 @@ class CMainParams : public CChainParams {
         consensus.vDeployments[Consensus::DEPLOYMENT_DIP0024].nThresholdMin = 2420;   // 60% of 4032
         consensus.vDeployments[Consensus::DEPLOYMENT_DIP0024].nFalloffCoeff = 5;      // this corresponds to 10 periods
 
+        // Deployment of Deployment of Basic BLS, AssetLock, EHF
```
Comment (knst) [inline src/chainparams.cpp:252]: AssetLock -> AssetLocks ?

File: test/functional/test_framework/test_framework.py [hunk 3]
```diff
@@ -929,6 +929,31 @@ def activate_dip0024(self, expected_activation_height=None):
         self.nodes[0].sporkupdate("SPORK_17_QUORUM_DKG_ENABLED", spork17_value)
         self.wait_for_sporks_same()
 
+    def activate_v19(self, expected_activation_height=None):
```
Comment (knst) [inline test/functional/test_framework/test_framework.py:932]: please, check this one: https://github.com/knst/dash/commit/33b6d629f8d6b8b81f8c6f09036d57c4deeb5614

File: src/chainparams.cpp (hunk 1 above)
Comment (ogabrielides) [reply src/chainparams.cpp:256]: Good catch!

File: test/functional/test_framework/test_framework.py (hunk 3 above)
Comment (ogabrielides) [reply test/functional/test_framework/test_framework.py:934]: Implemented your idea, but kept the sporks deactivation/reactivation only for dip0024

**Review by knst — APPROVED** [review]
utACK

**Comment by thephez** [comment]
LGTM :+1:

**Review by thephez — APPROVED** [review]
LGTM :+1:

**Review by UdjinM6 — APPROVED** [review]
utACK

File: src/consensus/params.h [hunk 4]
```diff
@@ -23,6 +23,7 @@ enum DeploymentPos {
     DEPLOYMENT_REALLOC, // Deployment of Block Reward Reallocation
     DEPLOYMENT_DIP0020, // Deployment of DIP0020, DIP0021 and LMQ_100_67 quorums
     DEPLOYMENT_DIP0024, // Deployment of DIP0024 (Quorum Rotation) and decreased governance proposal fee
+    DEPLOYMENT_V19, // Deployment of Basic BLS, AssetLocks, EHF
```
Comment (UdjinM6) [inline src/consensus/params.h:26]: nit:
```suggestion
    DEPLOYMENT_V19,     // Deployment of Basic BLS, AssetLocks, EHF
```

**Review by UdjinM6 — DISMISSED** [review]
re-utACK

**Review by thephez — DISMISSED** [review]
utACK

**Review by UdjinM6 — APPROVED** [review]
re-utACK

**Review by PastaPastaPasta — CHANGES_REQUESTED** [review]
NACK for merge now as discussed in our weekly call as I'm concerned about the `19999999999` start time for main net / testnet. I'm concerned that someone running this code when the DEPLOYMENT_V19 begins to be mined, that they may immediately hard fork off. This PR should be delayed until develop is 100% going to v19 instead of current potential v18.2

**Review by UdjinM6 — APPROVED** [review]
utACK

**Review by PastaPastaPasta — APPROVED** [review]
utACK for squash merge

**Pull request merged**